nomad/structs/structs.go

```diff
@@ -2579,6 +2578,24 @@ func (tg *TaskGroup) Validate(j *Job) error {
 	return mErr.ErrorOrNil()
 }
 
+// Warnings returns a list of warnings that may be from dubious settings or
+// deprecation warnings.
+func (tg *TaskGroup) Warnings(j *Job) error {
+	var mErr multierror.Error
+
+	// Validate the update strategy
+	if u := tg.Update; u != nil {
+		// Check the counts are appropriate
+		if u.MaxParallel > tg.Count {
+			mErr.Errors = append(mErr.Errors,
+				fmt.Errorf("Update max parallel count is greater than task group count (%d > %d). "+
+					"A destructive change would result in the replacement of all allocations.", u.MaxParallel, tg.Count))
```

Comment:
I feel like the key to mention is that they'll be replaced _at the same time_ resulting in downtime. The second line is always true regardless of whether max_parallel is > count or not... maybe add `simultaneous` to it?

> "A destructive change would result in the simultaneous replacement of all allocations.",